src/freenet/node/Node.java

```diff
@@ -1312,13 +1320,47 @@ public Short get() {
 
 					@Override
 					public void set(Short val) throws InvalidConfigValueException {
-						if(maxHTL < 0) throw new InvalidConfigValueException("Impossible max HTL");
+						if(val < 0) throw new InvalidConfigValueException("Impossible max HTL");
 						maxHTL = val;
 					}
 		}, false);
 
 		maxHTL = nodeConfig.getShort("maxHTL");
 
+		 class TrafficClassCallback extends StringCallback implements EnumerableOptionCallback {
+			 @Override
+			 public String get() {
+				 return trafficClass.name();
+			 }
+
+			 @Override
+			 public void set(String tcName) throws InvalidConfigValueException {
+				 try {
+					 trafficClass = TrafficClass.fromNameOrValue(tcName);
+				 } catch (IllegalArgumentException e) {
+					 throw new InvalidConfigValueException(e);
+				 }
+			 }
+
+			 @Override
+			 public String[] getPossibleValues() {
+				 ArrayList<String> array = new ArrayList<String>();
+				 for(TrafficClass tc : TrafficClass.values())
```

Comment:
there has to be a better way to do this... a git-pre-commit hook? a stylecheck thingy that travis would run?